src/github.com/matrix-org/dendrite/roomserver/storage/storage.go

```diff
@@ -380,6 +382,27 @@ func (d *Database) StateEntriesForTuples(
 	return d.statements.bulkSelectFilteredStateBlockEntries(stateBlockNIDs, stateKeyTuples)
 }
 
+// MembershipUpdater implements input.RoomEventDatabase
+func (d *Database) MembershipUpdater(roomID, targetUserID string) (updater types.MembershipUpdater, err error) {
+	err = common.WithTransaction(d.db, func(txn *sql.Tx) error {
+		var (
+			roomNID       types.RoomNID
+			targetUserNID types.EventStateKeyNID
+		)
+		roomNID, err = d.assignRoomNID(txn, roomID)
```

Comment:
ok. Just feels like a shame that you had to go and change `assignRoomNID` and the things it calls just so that you could pass in a transaction that you happened to have...